module-bastion.tf

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2017, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+locals {
+  bastion_public_ip = (var.create_bastion
+    ? one(module.bastion[*].public_ip)
+    : var.bastion_public_ip
+  )
+
+  bastion_image_id = (var.bastion_image_type == "custom"
+    ? var.bastion_image_id : element(tolist(setintersection([
+      lookup(local.image_ids, format("%s %s", var.bastion_image_os, split(".", var.bastion_image_os_version)[0]), null),
+      local.image_ids.nongpu, local.image_ids.x86_64,
+    ]...)), 0)
+  )
+}
+
+module "bastion" {
+  count          = var.create_bastion ? 1 : 0
+  source         = "./modules/bastion"
+  state_id       = random_id.state_id.id
+  compartment_id = local.compartment_id
+
+  # Bastion
+  assign_dns          = var.assign_dns
+  availability_domain = coalesce(var.bastion_availability_domain, lookup(local.ad_numbers_to_names, local.ad_numbers[0]))
+  image_id            = local.bastion_image_id
+  nsg_ids             = concat(var.bastion_nsg_ids, var.create_nsgs ? [module.network.bastion_nsg_id] : [])
+  public              = var.bastion_type == "public"
```

Comment:
Internal/inner module variable names should have the same. Please check the coding conventions for naming variables:

https://github.com/oracle-terraform-modules/terraform-oci-oke/blob/main/docs/codingconventions.adoc#31-case-type-files-names

Perhaps we should rename this to something like `assign_public_ip` ?

Reply:
1. Same question as with `worker_type` below - would this be better as an e.g. `assign_public_ip` boolean?
2. I don't personally feel it's necessary to require a full match between inner/outer of a sub-module so long as the difference is only a redundant prefix for inner, e.g. `nsg_ids` and `image_id` for `bastion_nsg_ids` and `operator_image_id` in their respective sub-modules aren't creating any confusion and reduce eye strain. If those sub-modules were to be used directly then having the prefix on every single var would be kind of weird, though no one else sees their spec here; it's just aesthetics for maintainers.

Reply:
Updated `worker_type` -> `worker_is_public` as discussed.

Reply:
Discussed and in agreement. Will updated coding conventions.